draft-ietf-mls-protocol.md

```diff
@@ -1439,6 +1479,39 @@ modification of its content MUST be reflected by a change in its
 signature. This allows other members to verify the validity of the KeyPackage
 at any time, particularly in the case of a newcomer joining the group.
 
+## KeyPackage Validation
+
+The validity of a KeyPackage needs to be verified at a few stages:
+
+* When a KeyPackage is downloaded by a group member, before it is used
+  to add the client to the group
+* When a KeyPackage is received by a group member in an Add, Update, or Commit
+  message
+* When a client joining a group receives KeyPackages for the other members of
+  the group in the group's ratchet tree
+
+The client verifies the validity of a KeyPackage using the following steps:
+
+* Verify that the credential in the KeyPackage is valid according to the
+  authentication service and the client's local policy.
```

Comment:
I suggest adding:
The authentication service validity check steps and client's local policy MUST be the same for all verification stages (KeyPackage download, Add, Update, Commit, or KeyPackage receipt).

Reply:
@bifurcation the updated changes look very good. I made a suggestion above. With that addition (or equivalent), it looks good to me.

Reply:
I would suggest do go slightly further and require that on top of the verification you do in case of an Add (i.e. asking the AS "Is this credential Ok?"), the AS should also answer the question "Is this credential OK as the successor to this previous credential?".

Reply:
Thanks @br-hale.  @kkohbrok I think that's covered down in the Update section.